examples/sst2/models.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2021 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A text classification model."""
+
+import collections
+import functools
+from typing import Any, Callable, Dict, Iterable, Optional, Sequence
+from absl import logging
+from flax import linen as nn
+import jax
+from jax import numpy as jnp
+import numpy as np
+
+Array = jnp.ndarray
+
+
+def sequence_mask(lengths: Array, max_length: int) -> Array:
+  """Computes a boolean mask over sequence positions for each given length.
+
+  Example:
+  ```
+  sequence_mask([1, 2], 3)
+  [[True, False, False],
+   [True, True, False]]
+  ```
+
+  Args:
+    lengths: The length of each sequence. <int>[batch_size]
+    max_length: The width of the boolean mask. Must be >= max(lengths).
+
+  Returns:
+    A mask with shape: <bool>[batch_size, max_length] indicating which
+    positions are valid for each sequence.
+  """
+  return jnp.arange(max_length) < jnp.expand_dims(lengths, 1)
+
+
+@jax.vmap
+def flip_sequences(inputs: Array, lengths: Array) -> Array:
+  """Flips a sequence of inputs along the time dimension.
+
+  This function can be used to prepare inputs for the reverse direction of a
+  bidirectional LSTM. It solves the issue that, when naively flipping multiple
+  padded sequences stored in a matrix, the first elements would be padding
+  values for those sequences that were padded. This function keeps the padding
+  at the end, while flipping the rest of the elements.
+
+  Example:
+  ```python
+  inputs = [[1, 0, 0],
+            [2, 3, 0]
+            [4, 5, 6]]
+  lengths = [1, 2, 3]
+  flip_sequences(inputs, lengths) = [[1, 0, 0],
+                                     [3, 2, 0],
+                                     [6, 5, 4]]
+  ```
+
+  Args:
+    inputs: An array of input IDs <int>[batch_size, seq_length].
+    lengths: The length of each sequence <int>[batch_size].
+
+  Returns:
+    An ndarray with the flipped inputs.
+  """
+  max_length = inputs.shape[0]
+  return jnp.flip(jnp.roll(inputs, max_length - lengths, axis=0), axis=0)
+
+
+class WordDropout(nn.Module):
+  """Applies word dropout to a batch of input IDs.
+
+  This is basically the same as `nn.Dropout`, but allows specifying the
+  value of dropped out items.
+  """
+  dropout_rate: float
+  unk_idx: int
+  deterministic: bool = False
+
+  @nn.compact
+  def __call__(self, inputs: Array):
```

Comment:
You could consider using `merge_params` here, which we added to avoid the confusion that users sometimes have related to whether `deterministic` is a constructor or apply argument: https://github.com/google/flax/blob/master/flax/linen/stochastic.py#L56

Reply:
Done :)